File: TheaterSuggestions/CSharp/SeatsSuggestions.Domain/SuggestionRequest.cs
using System.Collections.Generic;
using Value;

namespace SeatsSuggestions.Domain
{
    public class SuggestionRequest : ValueType<SuggestionRequest>
    {
        public PartyRequested PartyRequested { get; }
        public PricingCategory PricingCategory { get; }

        public SuggestionRequest(PartyRequested partyRequested, PricingCategory pricingCategory)
        {
            PartyRequested = partyRequested;
            PricingCategory = pricingCategory;
        }

        protected override IEnumerable<object> GetAllAttributesToBeUsedForEquality()
        {
            return new object[] {PartyRequested, PricingCategory};
        }

        public override string ToString()
        {
            return $"{PartyRequested}-{PricingCategory.ToString()}";
        }
    }
}

File: TheaterSuggestions/Java/SeatsSuggestionsDomain/src/main/java/com/baasie/SeatsSuggestionsDomain/SuggestionRequest.java
package com.baasie.SeatsSuggestionsDomain;

import lombok.Value;

@Value
public class SuggestionRequest {

    PartyRequested partyRequested;
    PricingCategory pricingCategory;

    public SuggestionRequest(PartyRequested partyRequested, PricingCategory pricingCategory) {
        this.partyRequested = partyRequested;
        this.pricingCategory = pricingCategory;
    }

    @Override
    public String toString() {
        return String.format("%s-%s", partyRequested, pricingCategory.toString());
    }

    public PricingCategory pricingCategory() {
        return pricingCategory;
    }

    public PartyRequested partyRequested() {
        return partyRequested;
    }
}